rpcs3/Emu/NP/clans_client.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <curl/curl.h>
+#include <pugixml.hpp>
+
+#include <Emu/Cell/Modules/sceNpClans.h>
+#include <Emu/NP/np_handler.h>
+#include <Emu/IdManager.h>
+
+inline const char* CLANS_ENTITLEMENT_ID = "NPWR00432_00";
+inline const char* CLANS_SERVICE_ID = "IV0001-NPXS01001_00";
+
+namespace clan
+{
+	enum class ClanManagerOperationType
+	{
+		VIEW,
+		UPDATE
+	};
+
+	enum class ClanRequestType
+	{
+		FUNC,
+		SEC
+	};
+
+	enum class ClanSearchFilterOperator : u8
+    {
+        Equal,
+        NotEqual,
+        GreaterThan,
+        GreaterThanOrEqual,
+        LessThan,
+        LessThanOrEqual,
+        Like,
+    };
+
+	enum class ClanRequestAction
+	{
+		GetClanList,
+		GetClanInfo,
+		GetMemberInfo,
+		GetMemberList,
+		GetBlacklist,
+		RecordBlacklistEntry,
+		DeleteBlacklistEntry,
+		ClanSearch,
+		CreateClan,
+		DisbandClan,
+		RequestMembership,
+		CancelRequestMembership,
+		AcceptMembershipRequest,
+		DeclineMembershipRequest,
+		SendInvitation,
+		CancelInvitation,
+		AcceptInvitation,
+		DeclineInvitation,
+		UpdateMemberInfo,
+		UpdateClanInfo,
+		JoinClan,
+		LeaveClan,
+		KickMember,
+		ChangeMemberRole,
+		RetrieveAnnouncements,
+		PostAnnouncement,
+		DeleteAnnouncement
+	};
+
+	class clans_client
+	{
+	private:
+
+
+		static size_t curlWriteCallback(void* data, size_t size, size_t nmemb, void* clientp);
+		SceNpClansError sendRequest(u32 reqId, ClanRequestAction action, ClanManagerOperationType type, pugi::xml_document* xmlBody, pugi::xml_document* outResponse);
```

Comment:
I just noticed that all of these functions are camelCase.
We try to keep new stuff in snake_case 😅
Better now than later

Reply:
Same with the variables